### PR TITLE
[air output] Add unit test to `experimental/output.py`

### PR DESCRIPTION
**Pull request by xwjiang2010**
## Why are these changes needed?
Add unit test to `experimental/output.py`.
Add some type hinting to methods.
Also fix a bug for rendering table.

These only cover basic unit testing. Will add separately some sort of integration test for the following scenarios:
* rich + tune
* non-rich + tune
* train

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by xwjiang2010** [comment]
Failing tests are not related.

**Review by krfricke — APPROVED** [review]
LGTM!

File: python/ray/tune/experimental/output.py [hunk 1]
```diff
@@ -335,7 +348,7 @@ def _best_trial_str(
 ):
     """Returns a readable message stating the current best trial."""
     # returns something like
-    # Current best trial: 18ae7_00005 with loss=0.5918508041056858 and parameters={'train_loop_config': {'lr': 0.059253447253394785}}. # noqa
+    # Current best trial: 18ae7_00005 with loss=0.5918508041056858 and params={'train_loop_config': {'lr': 0.059253447253394785}}. # noqa
```
[the remark this answers is not in the document]
Comment (xwjiang2010) [reply python/ray/tune/experimental/output.py:351]: Yes, dealing with it separately to limit the scope of this PR.

**Review by zhe-thoughts — APPROVED** [review]
This is test only. Approved for merging into master

**Pull request merged**